### PR TITLE
Remove light weight font rendering

### PR DESCRIPTION
**Pull request by globau**
Specifying the font-weight here is harmful when using fonts that support
Light or variable weights.

Signed-off-by: glob <byron@glob.com.au>

Notes: Remove light weight font rendering

---
This PR currently has no changelog labels, so will not be included in changelogs.

A reviewer can add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is, or add `Type: [enhancement/defect/task]` to the description and I'll add them for you.

Preview: https://pr7880--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

**Review by gsouquet — CHANGES_REQUESTED** [review]
It appears that there a couple more occurence of `font-weight: 300` in the codebase that would probably be suceptible to the same issue you raised. Could you update those ones too?

I looked at `@font-face` declaration and it appears that we're loading the following font weights: `400`, `500`, `600`, `700` for `Inter`. And `400` and `700` for `Inconsolata`

Will also loop in the design team to confirm this is a change we're comfortable making

**Review by gsouquet — APPROVED** [review]
Code looks good to me.
Please keep the design reviewers on. Without their approval we will not be abe to move this pull request forward

**Comment by janogarcia** [comment]
Unsure where those lighter fonts are being used, but, yeah, the current minimum weight should be 400.

**Review by janogarcia — COMMENTED** [review]
Are the affected components now inheriting the base font-weight of `400`?

**Comment by globau** [comment]
> Are the affected components now inheriting the base font-weight of `400`?

Excellent question!

As far as I can tell `mx_Dialog` always cascades to 400.

I can't find where `mx_CreateRoom` is used (no references to it under the `src` directory).

I'm not sure what a GroupView is so I can't verify manually in the Element client, but as far as I can tell (with my very limited knowledge here) the default weight for `mx_GroupView_header_shortDesc` _should be_ 400.

**Comment by t3chguy** [comment]
> I can't find where mx_CreateRoom is used (no references to it under the src directory).

It is no longer used, can we kill that entire file as part of this PR please.

**Comment by globau** [comment]
> > I can't find where mx_CreateRoom is used (no references to it under the src directory).
> 
> It is no longer used, can we kill that entire file as part of this PR please.

Removing an _potentially_ unused class seems to be unrelated to this issue, and I'd prefer if it were fixed in a separate PR.

**Comment by t3chguy** [comment]
Editing a class which does not cause your bug due to being unused is also unrelated to this issue though?

**Comment by janogarcia** [comment]
@globau Thanks for checking!

@t3chguy @gsouquet Can you please check/confirm if removing the declaration for `300` will inherit `400` for the cases where @globau is unsure how to test for?

**Comment by globau** [comment]
@janogarcia To move this along how about instead of removing the `font-weight: 300` lines I replace them with `font-weight: 400`?

That will guarantee correct rendering regardless of context, however it might mean clean up is required in a different PR due to possible redundancy.

**Comment by janogarcia** [comment]
@globau Sounds reasonable (ideally, if we could avoid those extra declarations, the better), but I'm afraid that would be up to @t3chguy and @gsouquet.

**Pull request closed without merging**
